authority/tls_test.go

```diff
@@ -370,9 +370,11 @@ ZYtQ9Ot36qc=
 	}
 }
 
-func TestAuthority_Renew(t *testing.T) {
```

Comment:
I like the idea of changing the `renew` tests to be `rekey` tests, because that's where all the logic lives. However, I think we should add a separate test function (`TestAuthority_Renew`) that verifies 1) if an error happens it's correctly returned, 2) on success, among other things the public key in the new cert is equivalent to the public key in the original cert. 

This doesn't have to be added as part of this PR. Just leaving a note for myself so I don't forget.